### ISSUE TITLE
[DO-40] Describe how to attach DOS to the ORT process

### ISSUE THREAD

**Issue by Etsija** [doubleopen-project/dos#88]
Needed:

* updated sequence diagram with realistic ORT process
* draw a crude architecture for attaching DOS as a new scanner
* how to store license conclusion information in relation to ORT datamodel
  * what is the structure of ORT license information?

### Our understanding of the current process of scanning an application with ORT

1. User runs ORT Analyzer for the whole packet -> analyzer-result.yml
2. User has a .ort.yml file in the application root folder with

```
excludes:
scopes:
- pattern: "devDependencies"
reason: "DEV_DEPENDENCY_OF"
comment: "Packages for development only."
```

1. User runs `ort scan -i ... -o ... --skip-excluded`. This in effect excludes all devDependencies from the scanning phase. For example, for mime_types project, the ORT Analyzer results in hundreds of packages. Out of these, only three packages are sent to ScanCode, and ScanCode returns the scan results for only these three packets

### How and where to interrupt this process with DOS concept?

By looking at analyzer-result.yml, it is not clear as to how the ort scan phase decides which of the packets to send to ScanCode, as there seems to be no such info on packet data in that file, to decide on exclusion or inclusion of a particular packet.

In practise, we need a way to "interrupt" the usual code analysing process in such a way that after the user has run the ORT Analyzer and sends the resulting analyzer-result.yml to DOS, DOS can decide which of the hundreds of packets are considered as candidates for scanning process.

In a nutshell: from ORT Analyzer results, how can the devDependencies (or, more generally, any packet the user wants to) be excluded from the scanning phase?

<sub>From [SyncLinear.com](https://synclinear.com) | [DO-40](https://linear.app/double-open/issue/DO-40/describe-how-to-attach-dos-to-the-ort-process)</sub>

**Comment by Etsija** [comment]
ORT scanner infrastructure discussion:

[https://github.com/oss-review-toolkit/ort/pull/5325](https://github.com/oss-review-toolkit/ort/pull/5325)

**Comment by Etsija** [comment]
A separate Linear team has been created to track the progress of this issue. The taam has its own issues (DOOR.xxx).

**Comment by Etsija** [comment]
I think this issue can be closed, as a separate team and project are tracking the progress of it.
